controllers/glanceapi_controller.go

```diff
@@ -1039,6 +1043,12 @@ func (r *GlanceAPIReconciler) generateServiceConfig(
 		"VHosts":       httpdVhostConfig,
 	}
 
+	// Only set EndpointID parameter when the Endpoint has been created and the
+	// associated ID is set in the keystoneapi CR
+	if len(endpointID) > 0 {
```

Comment:
I'm still not sure about this part: should I delay the config generation here if the `endpointID` is not present yet? The alternative is to update the secret when it's generated, but that might be an unnecessary patch and Pods rollout.
On the other hand this config is only required for Quota, so I'd like to reduce the scope of this code.

Reply:
I agree this seems a correct and safe check.
In case we have multiple APIs, let's say `default` and `default1`:

```yaml
glanceAPIs:
  keystoneEndpoint: default
  default1:
    type: single
   default:
    type: single
```
and `keystoneEndpoint: default`, the `keystoneendpoint` CR will be:

```
$ oc get keystoneendpoints
NAME                    STATUS   MESSAGE
glance-default-single   True     Setup complete
```

and the other API (`default1`), shouldn't be stuck because it waits for an endpoint that won't be set.
In case the endpoint is switched in the main CR (`keystoneendpoint: default1`), the previous endpoint CR is deleted, and a new one is created, with a different `endpointsID`.